lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -2345,16 +2345,15 @@ FuncInfo* PreVisitFunction(ParseNode* pnode, ByteCodeGenerator* byteCodeGenerato
             funcInfo->GetParsedFunctionBody()->SetUsesArgumentsObject(true);
             if (pnode->sxFnc.HasHeapArguments())
             {
-                bool doStackArgsOpt = !pnode->sxFnc.HasAnyWriteToFormals();
+                bool doStackArgsOpt = (!pnode->sxFnc.HasAnyWriteToFormals() || funcInfo->GetIsStrictMode());
```

Comment:
>GetIsStrictMode [](start = 89, length = 15)

Thanks for getting this in. Should see some benefits in node scenarios. BTW, do you mind adding couple of unit test?